microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/internal/core/MicroProfilePropertiesListenerManager.java

```diff
@@ -187,6 +197,29 @@ private boolean isJavaFile(IFile file) {
 			return JAVA_FILE_EXTENSION.equals(file.getFileExtension());
 		}
 
+		private boolean isConfigSource(IFile file) {
+			IPath path = file.getProjectRelativePath();
+			// we want the config update on the class path not the sources
+			if (path.segmentCount() <= 2

```

Comment:
It's annoying to hard code the output folder. I will remove this code and I will loop for each IConfigSourceProvider and I will add a new method `IConfigSource#isSameFile(IPath path)` and isSameFile should use https://github.com/eclipse/lsp4mp/blob/f9afc25b0ba3083251fd60cbca976bb3603d0750/microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/core/project/AbstractConfigSource.java#L70